### PR TITLE
Fix Doxygen warnings in ansi-c/ folder

### PR DESCRIPTION
**Pull request by johnnonweiler**
Delete one out-of-date (and unhelpful) comment and update other comments about functions and their arguments

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/ansi-c/c_typecheck_expr.cpp
Comment (hannes-steffenhagen-diffblue) [inline src/ansi-c/c_typecheck_expr.cpp]: > inserting implicit casts around parameters

:open_mouth: But _why_? I get that you're just documenting the existing behaviour, but I'm really not sure why a function called _check_ actually ends up modifying the expression it's supposedly checking...

**Review by NlightNFotis — APPROVED** [review]
LGTM

File: src/ansi-c/expr2c.cpp
Comment (hannes-steffenhagen-diffblue) [inline src/ansi-c/expr2c.cpp]: parenthesesis -> parentheses. [Parenthesis is also a word, as I just learned](https://en.oxforddictionaries.com/definition/parenthesis)

File: src/ansi-c/expr2c.cpp
Comment (hannes-steffenhagen-diffblue) [inline src/ansi-c/expr2c.cpp]: Hm, these comments will shut up doxygen but I'm not sure they're really improving the documentation here.

File: src/ansi-c/c_typecheck_expr.cpp
Comment (chrisr-diffblue) [reply src/ansi-c/c_typecheck_expr.cpp]: I think in CBMC we use 'typecheck' to mean a bit more than what a typical compiler would mean by 'typecheck'... It seems to be a sort of 'typecheck and canonicalise' operation - which, while I agree the naming is not great, is not entirely an unreasonable thing to do.

File: src/ansi-c/c_typecheck_initializer.cpp
Comment (hannes-steffenhagen-diffblue) [inline src/ansi-c/c_typecheck_initializer.cpp:354]: Why is this being removed with no replacement?

File: src/ansi-c/expr2c.cpp
Comment (johnnonweiler) [reply src/ansi-c/expr2c.cpp]: 👍

**Review by chrisr-diffblue — APPROVED** [review]
Approving because it cleans up some Doxygen warnings. I agree that it'd be nice to further improve the docs in a couple of places, but this PR clearly states 'Fix Doxygen warnings in ...' so I'm not going to stamp my feet and complain.

File: src/ansi-c/expr2c.cpp
Comment (johnnonweiler) [reply src/ansi-c/expr2c.cpp]: The main point of this PR is to shut up Doxygen.  We've set Doxygen to give warnings if you document some parameters and not others, and sometimes that means that we have documentation for parameters which is not especially helpful.

File: src/ansi-c/c_typecheck_initializer.cpp
Comment (johnnonweiler) [reply src/ansi-c/c_typecheck_initializer.cpp:354]: As explained in the commit message, 
> Obviously, it would be
better to add some up-to-date documentation, but until someone has time
to do that, this removes the unhelpful comment and stops Doxygen
generating 3 warnings.

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]
Comments have been addressed, I'm assuming spelling errors will be fixed before merge

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 38310c5ba2a82c2ca1c29c4e4f1438ab1a4a7f9c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91219296

**Comment by johnnonweiler** [comment]
I've fixed the spelling mistake.  This is now just waiting for a code owner review from @kroening or @tautschnig.

**Review by peterschrammel — APPROVED** [review]

File: src/ansi-c/c_typecheck_expr.cpp
Comment (tautschnig) [reply src/ansi-c/c_typecheck_expr.cpp]: Just as a comment on the proposed new wording: we insert casts, which in fact makes them no longer _implicit_.

File: src/ansi-c/expr2c.cpp
Comment (tautschnig) [reply src/ansi-c/expr2c.cpp]: How about wording this as "The previously computed string denoting the array symbol" - if that's any clearer @hannes-steffenhagen-diffblue?

**Review by tautschnig — APPROVED** [review]

**Comment by johnnonweiler** [comment]
This has now been rebased, and updated based on suggestions from @tautschnig.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 2e70949237eae792f731e9abbd67b1abbbda242a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91500827
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c4a52fd939805bfd361b0d6cdce562a4a0824f9f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91505942

**Pull request merged**
